src/application/common/logger.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2017 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as Logger from 'bunyan';
+
+const ILogger = Symbol("ILogger");
+
+interface ILogger {
+    addStream(stream: Logger.Stream): Promise<void>;
```

Comment:
Well that was my first I idea I thought let's just expose bunyan. Thus my proxy approch etc.. 
But looking at your review I now think this was not the right approach especially since we need to control what we send as objects over JSON-RPC.. 

But I think we would like at least to:
- child : So that extensions can log in their scope
- level: Get and Set so that we can change the level at runtime

The streams I don't think we can send that over json-rpc given that it's using a nodeJS stream object.
Same with the serializers they're a complex object.

So we could expose:
- log level functions
- child creation
- logging functions 
And the logging functions would be string only over json-rpc to avoid sending random objects over json-rpc, 
But we could wrap around a format string interface to be used by the user.